common/src/main/java/org/opensearch/ml/common/agent/LLMSpec.java

```diff
@@ -19,7 +20,7 @@
 import static org.opensearch.core.xcontent.XContentParserUtils.ensureExpectedToken;
 import static org.opensearch.ml.common.utils.StringUtils.getParameterMap;
 
-
+@EqualsAndHashCode
```

Comment:
Just curious to know why do we need this?

Reply:
Mainly to make it easier to write UT to assert equals